### PR TITLE
Additional examples and recipes for various testing scenarios

### PR DESCRIPTION
**Pull request by DanielJette**
# What's in this PR?

I have added tests to the Sample app demonstrating the majority of the `ScreenshotRule` functionality. In addition, I have expanded the `RECIPE` document to describe how to set up various advanced testing scenarios.

## Tophat

You can read the recipe book [here](https://github.com/Shopify/android-testify/blob/beta1-samples/RECIPES.md)

### CODE REVIEW

File: RECIPES.md [hunk 1]
```diff
@@ -183,3 +298,75 @@ Use the `setOrientation` method to select between portrait and landscape mode.
 ```
 
 ---
+
+## Debugging with the Layout Inspector
```
Comment (lwise) [inline RECIPES.md:302]: I didn't know we had this feature - very cool!

File: RECIPES.md [hunk 2]
```diff
@@ -183,3 +298,75 @@ Use the `setOrientation` method to select between portrait and landscape mode.
...
+                .setLayoutInspectionModeEnabled(true)
+                .assertSame()
+    }
+```
+
+---
+
+## Selecting an alternative capture method
+
+Testify provides three bitmap capture method. Each method will capture slightly different results based primarily on API level.
```
Comment (lwise) [inline RECIPES.md:320]:
```suggestion
Testify provides three bitmap capture methods. Each method will capture slightly different results based primarily on API level.
```

File: Sample/src/androidTest/java/com/shopify/testify/sample/ScreenshotRuleExampleTests.kt [hunk 3]
```diff
@@ -71,6 +78,162 @@ class ScreenshotRuleExampleTests {
...
+            }
+        }.assertSame()
+    }
+
+    /**
+     * Demonstrates some diagnostic information (test name) that's available to assist logging
+     */
+    @ScreenshotInstrumentation
+    @Test
+    fun diagnosticInfo() {
```
Comment (lwise) [inline Sample/src/androidTest/java/com/shopify/testify/sample/ScreenshotRuleExampleTests.kt:103]: Is this screenshot supposed to be blank?

Comment (DanielJette) [reply Sample/src/androidTest/java/com/shopify/testify/sample/ScreenshotRuleExampleTests.kt:103]: Yeah. It doesn't actually load anything to test.
I should probably move this to a non-screenshot test, now that you point this out. It's silly the way it is

Comment (DanielJette) [reply Sample/src/androidTest/java/com/shopify/testify/sample/ScreenshotRuleExampleTests.kt]: I decided to remove this test entirely. It didn't provide any value

**Pull request merged**